huff_utils/src/ast.rs

```diff
@@ -532,6 +560,25 @@ impl MacroDefinition {
                         ty: IRByteType::Bytes(Bytes(opcode_str)),
                         span: statement.span.clone(),
                     });
+                    // If the opcode is a push, we need to consume the next statement, which must be
+                    // a literal as checked in the parser
+                    if o.is_push() {
+                        match statement_iter.next() {
+                            Some(Statement { ty: StatementType::Literal(l), span: _ }) => {
+                                let hex_literal: String = bytes32_to_string(l, false);
+                                inner_irbytes.push(IRBytes {
+                                    ty: IRByteType::Bytes(Bytes(hex_literal)),
+                                    span: statement.span.clone(),
+                                });
+                                statement_iter.next();
```

Comment:
Any specific reason for consuming the statement after the literal? This looks like it should be removed, as `push<n>` should only consume the literal.

Example code that produces unexpected output:
```
#define macro MAIN() = {
    push1 0x00
    0x00
}
```
Runtime expected:
```js
0x60006000
```
Runtime produced:
```js
0x6000
```

Reply:
ah great catch!!